File: typescript/src/ships/internal/capacitor-recharge.ts
/**
 * Scale a distributor's rated four-pip recharge to one pip allocation.
 *
 * Reference model: EDCD/Coriolis. Credit and licence terms are in
 * [ATTRIBUTIONS.md](https://github.com/DarkSession/Elite-Dangerous-Almanac/blob/main/ATTRIBUTIONS.md).
 *
 * @param ratedRecharge - Recharge at four pips, in energy per second.
 * @param pips - Validated capacitor allocation in `[0, 4]`.
 * @returns Recharge at `pips`, in the same units as `ratedRecharge`.
 * @internal
 */
export function capacitorRechargeAtPips(ratedRecharge: number, pips: number): number {
    return ratedRecharge * Math.pow(pips / 4, 1.1);
}

File: dotnet/src/EliteDangerousAlmanac/Ships/Internal/CapacitorRecharge.cs
using System;

namespace EliteDangerousAlmanac.Ships.Internal;

/// <summary>The curve a distributor's rated four-pip recharge is scaled along.</summary>
/// <remarks>
/// Reference models: EDCD/Coriolis for SYS and WEP, and EDSY for ENG. See <c>ATTRIBUTIONS.md</c>
/// for credit and licence terms.
/// </remarks>
internal static class CapacitorRecharge
{
    /// <summary>Scales a rated four-pip recharge to one pip allocation.</summary>
    /// <param name="ratedRecharge">The recharge at four pips, in energy per second.</param>
    /// <param name="pips">The capacitor allocation, from zero through four.</param>
    /// <returns>The recharge at the allocation, in the same units as the rated recharge.</returns>
    internal static double AtPips(double ratedRecharge, double pips) =>
        ratedRecharge * Math.Pow(pips / 4, 1.1);
}
